src/init.cpp

```diff
@@ -1559,7 +1559,7 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
                 chainman.m_total_coinstip_cache = nCoinCacheUsage;
                 chainman.m_total_coinsdb_cache = nCoinDBCache;
 
-                UnloadBlockIndex(node.mempool.get());
+                UnloadBlockIndex(node.mempool.get(), &chainman);
```

Comment:
in commit bc872e44e29f7108720072a7d37ce06fb069ba4f: Doesn't it make more sense to pass it as `this` (i.e. make it a member function)?

I've done that in #19909 (commit facd88a77189f6953d28a6743e0c68175bc18f9e)

Conceptually it makes little sense for chainman to be optional. Why would you want to run a node with validation disabled? So, if you don't like to make it a member function, I'd say to at least pass it as reference.

Reply:
I will take another look at your PR, but I think in this particular instance, we should pass `chainman` as a reference, simply because `UnloadBlockIndex` touches so many validation globals that are not owned by `chainman`. Perhaps in the future when we move those globals into `chainman` or some other validation-container class, it would make more sense to do so.

I was worried about callers (to `UnloadBlockIndex`) who only have a pointer to `chainman`, but the only caller I see that only has a pointer to `chainman` is `~TestingSetup`, and it already looks like it makes the assumption that `m_node.chainman` is not null (see the `m_node.chainman->Reset()` line in `~TestingSetup`). Will make the change and take a look at your PR too.

Reply:
I also have a gut feeling that UnloadBlockIndex should eventually be a member function of chainman, but agree that we don't need to do everything in one PR.